packages/synapse-react-client/src/components/SynapseTable/usePrefetchTableData.ts

```diff
@@ -164,40 +167,45 @@ function usePrefetchEntityData(entitiesToPrefetch: ReferenceList) {
   })
 }
 
-function usePrefetchUserGroupHeaderData() {
-  const data = useAtomValue(tableQueryDataAtom)
+function usePrefetchUserGroupHeaderData(rowSet: RowSet) {
```

Comment:
Updated to pass RowSet as an argument, and use queryContext + react-query instead of Jotai. No logical changes